analysis/SimulationDataset.py

```diff
@@ -107,47 +106,53 @@ def SampleOneEvent( self, EnergyResolution=0.0, TimeResolution=0.0 ):
   def size( self ):
     return self.totalDecays
 
-def FindHitRadius( Event, DetectorRadius ):
+def FindHitRadius( Event, DetectorRadius):
   if len( Event ) != 2:
     return -1.0
 
+  #check event ID from both photons 
+  hasSameEventID = False
+  if (Event[0][0] == Event[1][0]) :
+    hasSameEventID = True
+
   # Calculate delta phi
-  phi1 = Event[0][4]
-  phi2 = Event[1][4]
+  phi1 = Event[0][5]
+  phi2 = Event[1][5]
   deltaPhi = phi1 - phi2
   while deltaPhi > math.pi:
     deltaPhi -= 2.0 * math.pi
   while deltaPhi < -math.pi:
     deltaPhi += 2.0 * math.pi
 
   if deltaPhi < 0.0:
-    return -DetectorRadius * math.cos( deltaPhi/2.0 )
+    return -DetectorRadius * math.cos( deltaPhi/2.0 ), hasSameEventID
```

Comment:
As mentioned before, surely better to return the "hasSameEventID" from its own dedicated method.

Reply:
Added a dedicated function